examples/common/src/main.rs

```diff
@@ -88,12 +112,8 @@ fn sign_transaction(
     private_key_wif: String,
     serialized_transaction: Vec<u8>,
     source_address: String,
-) -> Vec<u8> {
-    let private_key = PrivateKey::from_wif(&private_key_wif).expect("Invalid private key WIF");
-    let tx: Transaction =
-        deserialize(serialized_transaction.as_slice()).expect("Invalid transaction");
-    let source_address = Address::from_str(&source_address).expect("Invalid source address");
-    example_common::sign_transaction(tx, private_key, source_address).serialize()
+) -> Result<Vec<u8>, SignTransactionError> {
+    sign_transaction_(private_key_wif, serialized_transaction, source_address)
```

Comment:
What's the point of having private functions that do the same as the public functions?